pkg/worker/worker.go

```diff
@@ -297,7 +300,13 @@ func (w *Worker) setupTests(ex executor.Executor, workDir, repoDir string) (bool
 	//Remove any existing workdir of same name, ussually due to termination of jobs
 	status, err := w.runCommand(true, ex, "", []string{"rm", "-rf", workDir})
 	if err != nil {
-		w.handleCommandError(ex.GetTags(), err)
+		// if failure occurred due to `Device or resource busy` skip the cleanup
+		if strings.Contains(fmt.Sprint(err), "Device or resource busy") {
+			status = true
+			klog.V(2).Infof("skipping cleanup step")
```

Comment:
If we skip it here, where are we deleting the dir then? This might consume lots of resources, AFAIU these type of files are very big and might take up full memory very quickly. @mohammedzee1000 Could you please share your inputs as well.

Reply:
@prietyc123 test usually passes, if it test pass ci-firewall deletes all the file under `amqp...` dir
we can also add if the test fails we will remove it. that will also help having a clean env for tests

Reply:
We should actually be careful about cleanup here as there is the possibility of race condition which is likely what is causing this problem. A possible solution could be retry cleanup N times before giving up.

Another thing we could consider is building some logic to avoid race condition entirely. Maybe prevent 2 jobs from acting on the same location at once, like maybe a `lock file` which cleanup checks for and basically waits until this file disappears

Reply:
> Maybe prevent 2 jobs from acting on the same location at once, like maybe a lock file which cleanup checks for and basically waits until this file disappears.

+1 for using `lock file` concept. I think this will remove race condition completely.

Reply:
> > Maybe prevent 2 jobs from acting on the same location at once, like maybe a lock file which cleanup checks for and basically waits until this file disappears.
> 
> +1 for using `lock file` concept. I think this will remove race condition completely.

If we are going to fo with lock file approach then, this is what I would recommend

 - Add a couple of functions to the Exector interface say `Lock` and `IsLocked`.
 - Implement this for both executors (Local and Node) to check if file exists and create it respectively. 
 - Rotate on `IsLocked` before attempting rm -rf. 
 - Once the rotation breaks, if due to timeout error out else `Lock` and proceed